src/core/qgsmapsettings.h

```diff
@@ -102,6 +102,16 @@ class CORE_EXPORT QgsMapSettings
      */
     void setExtent( const QgsRectangle &rect, bool magnified = true );
 
+    /**
+     * \since QGIS 3.10
+     */
+    double extentBuffer() const;
+
+    /**
+     * \since QGIS 3.10
```

Comment:
This will need some more docstrings to explain what this does in detail

Reply:
Sure. This is still WIP ;)

Reply:
Done.

Reply:
Can you also describe what units this is in?

Reply:
Good call. New commit added for this.

Reply:
And for the record I also added a tooltip to describe the TileBuffer setting in the UI.